chia/types/blockchain_format/program.py

```diff
@@ -129,12 +140,26 @@ def _run(self, max_cost: int, flags: int, args: Any) -> Tuple[int, Program]:
         return cost, Program.to(r)
 
     def run_with_cost(self, max_cost: int, args: Any) -> Tuple[int, Program]:
-        return self._run(max_cost, 0, args)
+        # when running puzzles in the wallet, default to enabling all soft-forks
+        # as well as enabling mempool-mode (i.e. strict mode)
+        default_flags = (
+            ENABLE_SOFTFORK_CONDITION
+            | ENABLE_BLS_OPS_OUTSIDE_GUARD
+            | ENABLE_FIXED_DIV
+            | AGG_SIG_ARGS
+            | ENABLE_MESSAGE_CONDITIONS
+            | DISALLOW_INFINITY_G1
+            | MEMPOOL_MODE
+        )
+        return self._run(max_cost, default_flags, args)
 
     def run(self, args: Any) -> Program:
```

Comment:
I did a bad job in evolving this `run` function. In hindsight, I think we should consider this:

```suggestion
    def run(self, args: Any, /, max_cost=INFINITE_COST, flags=DEFAULT_FLAGS) -> Program:
```

Then we can move `DEFAULT_FLAGS` to the top level and just define `run_with_cost` in terms of `run` (and maybe even deprecate it).

Reply:
Maybe we don't even need the `._run` method anymore.